### PR TITLE
Retention Configuration (Set, Get) with Args, Response objects

### PR DESCRIPTION
**Pull request by BigUstad**
Examples, Functional Tests

### CODE REVIEW

**Comment by sinhaashish** [comment]
@BigUstad  can you please resolve the conflict.

**Comment by BigUstad** [comment]
> @BigUstad can you please resolve the conflict.

Conflicts resolved. PTAL @sinhaashish

**Review by sinhaashish — APPROVED** [review]
LGTM

**Review by krisis — APPROVED** [review]

File: Minio/DataModel/ObjectOperationsArgs.cs [hunk 1]
```diff
@@ -607,4 +611,141 @@ public override RestRequest BuildRequest(RestRequest request)
...
+
+        public override void Validate()
+        {
+            base.Validate();
+            if (this.RetentionUntilDate.Equals(default(DateTime)) && this.RetentionValidDays < 0)
+            {
+                throw new InvalidOperationException("Retention Period is not set. Please set " +
+                        nameof(RetentionUntilDate) + " or " + nameof(RetentionValidDays));
+            }
+            if (!this.RetentionUntilDate.Equals(default(DateTime)) && (this.RetentionUntilDate - DateTime.Today).TotalDays <= 0.0)
```
Comment (kannappanr) [inline Minio/DataModel/ObjectOperationsArgs.cs:637]: This check is wrong. Please remove the days argument. Not sure why this was added here.

**Review by kannappanr — CHANGES_REQUESTED** [review]
Please fix the validation and remove days as a parameter

File: Minio/DataModel/ObjectOperationsArgs.cs (hunk 1 above)
Comment (BigUstad) [reply Minio/DataModel/ObjectOperationsArgs.cs:636]: Days as parameter removed. Only date is retained

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**